src/utils/mod.rs

```diff
@@ -1013,7 +1300,33 @@ pub fn create_unrestricted_signing_rsa_public(
         .with_name_alg(TPM2_ALG_SHA256)
         .with_object_attributes(object_attributes)
         .with_parms(PublicParmsUnion::RsaDetail(rsa_parms))
-        .build() // should not fail as we control the params
+        .build()
+}
+
+/// Create the TPM2B_PUBLIC structure for an ECC unrestricted signing key.
+///
+/// * `scheme` - Asymmetric scheme to be used for signing
+/// * `curve` - identifier of the precise curve to be used with the key
+pub fn create_unrestricted_signing_ecc_public(
+    scheme: AsymSchemeUnion,
+    curve: EllipticCurve,
+) -> Result<TPM2B_PUBLIC> {
```

Comment:
Is `TPM2B_PUBLIC` the last structure that we use and do not abstract? (Just for my information)

Reply:
Yeah, `TPM2B_PUBLIC` is really deeply nested and convoluted - I guess it could be made up of a hierarchy of Rust native types, but it would be a large effort to allow all the possibilities that the original interface allows, while making it easier for a developer to build a structure that just works™️.

Reply:
Got it.